src/http_connection.cpp

```diff
@@ -508,6 +508,9 @@ void http_connection::on_resolve(error_code const& e
 		m_endpoints.emplace_back(addr, m_port);
 
 	if (m_filter_handler) m_filter_handler(*this, m_endpoints);
+
+	if (m_abort) return;
```

Comment:
is the reason you check `m_abort` here to catch it when the filter handler function triggers the abort?
It would seem reasonable to check it at the top of the function otherwise.

Reply:
from your description, I would also have expected this check to happen *before* `m_start_time` being set, since that resets the timeout.

Reply:
fixed